clippy_lints/src/restricted_scripts.rs

```diff
@@ -0,0 +1,254 @@
+use clippy_utils::diagnostics::span_lint;
+use rustc_data_structures::fx::FxHashSet;
+use rustc_hir::{Pat, PatKind};
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_session::{declare_tool_lint, impl_lint_pass};
+use unicode_script::{Script, UnicodeScript};
+
+declare_clippy_lint! {
+    /// **What it does:** Checks for usage of unicode scripts other than explicitly allowed
+    /// by the lint config.
+    ///
+    /// This lint doesn't take into account non-text scripts such as `Unknown` and `Linear_A`.
+    /// It also ignores the `Common` script type.
+    /// While configuring, be sure to use official script names from [the list of supported scripts][supported_scripts]
+    /// and replace spaces with undescores.
+    ///
+    /// For example, `Syloti Nagri` must be represented as `"Syloti_Nagri`".
+    ///
+    /// [supported_scripts]: http://www.unicode.org/standard/supported.html
```

Comment:
Well, it looks like it'll be more complicated to implement. I can see, for example, the alias `Katakana_Or_Hiragana`, which represents *two* scripts rather than one.
In theory it could've implemented through `ScriptExtension`s, but I don't see a way to construct it manually, which means that we'll have to return `Vec<Script>`, which is less efficient.

Do you find it significant to stick to the aliases?

Reply:
> Well, it looks like it'll be more complicated to implement. I can see, for example, the alias `Katakana_Or_Hiragana`, which represents _two_ scripts rather than one.
> In theory it could've implemented through `ScriptExtension`s, but I don't see a way to construct it manually, which means that we'll have to return `Vec<Script>`, which is less efficient.

Okay, this had me puzzled for a bit but I finally figured it out - it is in fact one script but it is no longer used in the latest version of Unicode. I think we can just ignore special cases like this. I looked at the source of the `unicode-script` crate and it is definitely using the "Alias" values.

> Do you find it significant to stick to the aliases?

Yes, it is better to use a pre-existing naming convention.

Reply:
FYI I learned about Katakana_Or_Hiragana being a special case on [this page](https://unicode.org/reports/tr31/).

Reply:
Oh, if you're looking for handling that special case you should use the AugmentedScriptSet property from unicode-security instead

Reply:
I added a note with a link to the page about it being a special case instead. I think that it's better than adding one more dependency.

Reply:
There are lots of scripts that are not used in current Unicode. Katakana_Or_Hiragana is just one of them. I don't think it's anything we need to document or worry about.

Reply:
OK then, removed 😅